docs/gss.md

```diff
@@ -0,0 +1,305 @@
+* `rule`: The rule that was violated.
+* `message`: The i18n key for the validation error.
+
+Although you may supply i18n messages that follow whatever conventions you choose, the default validation error messages
+included with this package support the use of variables that come from the error. Continuing the above example, let's
+look at how the second failure might be used with an i18n message template.  That error looks like:
+
+```javascript
+// This would ordinarily be delivered by the validator, but is presented here in isolation for illustration purposes.
+var error = {
+    "dataPath": ["deep", "booleanField"],
+    "schemaPath": ["properties", "deep", "properties", "booleanField", "type"],
+    "rule": { "type": "boolean" },
+    "message": "validation-invalid-field-type"
+};
+
+// This is simplified for illustration purposes, this would nearly always be delivered as part of a more complex bundle.
+var messages = {
+    "validation-invalid-field-type": "The value supplied should be a(n) %rule.type."
+};
+
+var message = fluid.stringTemplate(messages[error.message], error); // The value supplied should be a(n) boolean.
+```
+
+## Schema-validated components
+
+A schema validated component is one whose structure is required to match its "schema" options, which is defined using
+GSS.  A schema validated component will throw an error during its instantiation if there are validation errors.  Among
+many other things, this allows authors to write "dumber" components that can safely assume that all required options
+exist, and that all options are set to a useful value.
+
+### `gpii.schema.component`
+
+The `gpii.schema.component` grade is the base grade for all schema-validated components.  It includes a base schema that
+describes the standard elements of any Fluid component, such as `gradeNames`, `events`, `listeners`, and (sub)
+`components`.
+
+### Creating your own schema-validated grades.
+
+Let's say that you want to create a new component, that has a required option `requiredOption`, which should be a
+non-empty string.  Your grade definition might look like:
+
+```javascript
+fluid.defaults("my.schema.grade", {
+    gradeNames: ["gpii.schema.component"],
+    schema: {
+        properties: {
+            options: {
+                properties: {
+                    requiredOption: {
+                        required: true,
+                        type: "string",
+                        minLength: 1
+                    }
+                }
+            }
+        }
+    }
+});
+```
+
+Someone using or extending your grade must ensure that `options.requiredOption` exists and meets the specified criteria.
+Otherwise, the component will not instantiate, and an error like the following will be thrown:
+
+```text
+Component does not match its own schema, aborting component creation:
+    options -> requiredOption:  This value is required.
+```
+
+## Schema-validated model components
```

Comment:
Should add a stub docs plus ref to the dedicated docs here

Reply:
Done.